app/assets/javascripts/surveys.js

```diff
@@ -320,6 +322,27 @@
       }
     },
 
+    attachGa4FormCompleteElement: function (emailSurveyElement, error) {
+      // When the email survey form is submitted, this attaches a HTML element with our GA4 auto tracker to the DOM.
+      // The module is then started - the module will send data-ga4-auto object to the dataLayer.
+      // Doing it this way prevents us from calling and having to maintaining GA4 JS/cookie consent in this repo.
+      var emailSurveyHeading = document.getElementById('survey-title').textContent.trim()
+      var emailSurveyText = error ? document.getElementById('email-survey-post-failure') : document.getElementById('#email-survey-post-success')
+      emailSurveyText = emailSurveyText.textContent.trim()
+      var span = document.createElement('span')
+      span.setAttribute('data-module', 'ga4-auto-tracker')
+      span.setAttribute('data-ga4-auto', JSON.stringify({
+        event_name: 'form_complete',
+        action: 'complete',
+        type: 'survey banner',
+        text: emailSurveyText,
+        section: emailSurveyHeading,
+        tool_name: emailSurveyHeading
+      }))
+      emailSurveyElement.appendChild(span)
+      window.GOVUK.modules.start(emailSurveyElement)
```

Comment:
This looks good but I'm wondering if we should think about our modules in this context a bit more. This isn't a criticism of the work you've done here, more something maybe for us to think about in general.

It feels slightly clumsy to create an element, attach attributes to it, then call start modules on it, when perhaps we could be calling the module directly and just pass it what it needs? Maybe a bit more like we do when we run tests...

```Javascript
var attributes = { 'ga4': 'attributes' }
var module = new GOVUK.Modules.Ga4AutoTracker({ attributes: attributes })
module.init()
```

Reply:
Thanks @andysellick - that's a good idea, I think we should implement that :+1: